feat: rename set command's `--parent` flag to `--parents`

jrz opened the pull request:
# Summary

Changes --parent to --parents. Updated docs etc.
It adds a fallback --parent, but will warn about the change. 
Note that the old option still appears in --help (Hence the Hidden: false in cli.go. That can be set to true, or just removed.)

<!-- Provide a general description of the code changes in your pull request. -->

Fixes: List issue numbers here
#2010 Consider renaming --parent to --parents

## Other Information
Please check if you want the Hidden: true or not.

I noticed tests fail, but the same tests fail from a fresh master.
I couldn't really think of a test, as the option itself is only used as -p in the test.

Lastly, in the contribute/core.html I noticed the instructions said bats test/list_commands.bash instead of bats test/list_commands.bats

Stratus3D commented on CHANGELOG.md:
```diff
@@ -8,7 +8,7 @@
 
 ## [0.16.6](https://github.com/asdf-vm/asdf/compare/v0.16.5...v0.16.6) (2025-03-21)
-
+* rename --parent argument to --parents ([#2010](https://github.com/asdf-vm/asdf/issues/2010))
```
Please revert this change. CHANGELOG.md entries are added automatically.

jrz commented:
Reverted, and squashed some of the commits.
The last one could be a separate chore.

First commit is for global renaming + a bit or rewording
Second commit is the actual code
